include/envoy/http/async_client.h

```diff
@@ -131,34 +131,83 @@ class AsyncClient {
 
   virtual ~AsyncClient() {}
 
+  /**
+   * A structure to hold the options for AsyncStream object.
+   */
+  struct StreamOptions {
+    StreamOptions() {}
```

Comment:
Do you need those constructors? You should be able to do `StreamOptions{}` to do most of them.
https://en.cppreference.com/w/cpp/language/aggregate_initialization

Reply:
Not sure how to do this.

e.g. After removed all constructors.   Not sure how to initialize it with a specific field and other fields using default. For example, for timeout field,  if I do this:

send(, , RequestOptions{.timeout = xyz});

This will not compile

Reply:
The point is not to use that much default values, You should be able to do: `RequestOptions{{std::chrono::milliseconds(1000)}}`
`RequestOptions{{std::chrono::milliseconds(1000)}, true}` 
`RequestOptions{{std::chrono::milliseconds(1000)}, true, false}`

named initialization won't compile until C++20

Reply:
Hmm,  tried. It still doesn't work.

Since  RequestOption is derived from StreamOptions, This requires 
// aggregate in C++17
feature

I tried this:

 struct AA { int a, b = 1; };
 struct BB : AA { int c; };

BB b{{1, 2}, 3};

this will not compile

Reply:
A more readable, but less efficient alternative, would be to have a bunch of `static` builder methods that take a `StreamOptions` and return a `StreamOptions`, so you could just write:
```
StreamOptions().setXff(false).setTimeout(std::chrono::milliseconds(1000)).setBufferBodyForRetry(true)
```

Reply:
As long as we are bikeshedding on this, +1 to builder, or just have a single default constructor and have the caller stack allocate it and modify the params before calling?